File: javascript algorithms/검색,정렬/bubbleSort.js
// 시간복잡도 O(n2)

export function swap(array, index1, index2) {
  let temp = array[index1];

  array[index1] = array[index2];
  array[index2] = temp;
}

function bubbleSort(array) {
  for (let i = 0; i < array.length; i++) {
    for (let j = 0; j <= i; j++) {
      if (array[j] > array[j + 1]) {
        swap(array, j, j + 1);
      }
    }
  }
  return array;
}

bubbleSort([6, 1, 2, 5, 3, 7, 8]);

File: algorithm/검색,정렬/bubbleSort.py
import random


def bubblesort(data):
    swap = False
    for index in range(len(data)-1):
        for index2 in range(len(data)-index-1):
            if data[index2] > data[index2 + 1]:
                data[index2], data[index2 + 1] = data[index2 + 1], data[index2]
                swap = True

        if swap == False:
            break
    return data


data_list = random.sample(range(100), 50)
bubblesort(data_list)

# 시간분석
# O(n2) 이중 for
